checked_message_test.go

```diff
@@ -60,3 +60,143 @@ func TestCheckedMessageIsSingleUse(t *testing.T) {
 		assert.Equal(t, expected, buf.Lines(), "Expected re-using a CheckedMessage to log an error.")
 	})
 }
+
+func TestCheckedMessage_Chain(t *testing.T) {
+	withJSONLogger(t, opts(InfoLevel), func(logger Logger, buf *testBuffer) {
+		loga := logger.With(String("name", "A"))
+		logb := logger.With(String("name", "B"))
+		logc := logger.With(String("name", "C"))
+
+		for i, tt := range []struct {
+			cms        []*CheckedMessage
+			expectedOK bool
+			desc       string
+			expected   []string
+		}{
+			// not-ok base cases
+			{
+				cms:        []*CheckedMessage{nil},
+				expectedOK: false,
+				desc:       "nil",
+				expected:   []string{},
+			},
+			{
+				cms: []*CheckedMessage{
+					nil,
+					loga.Check(DebugLevel, "nope"),
+				},
+				expectedOK: false,
+				desc:       "nil, A decline",
+				expected:   []string{},
+			},
+			{
+				cms: []*CheckedMessage{
+					nil,
+					loga.Check(DebugLevel, "nope"),
+					logb.Check(DebugLevel, "nope"),
+				},
+				expectedOK: false,
+				desc:       "nil, A and B decline",
+				expected:   []string{},
+			},
+
+			// singleton ok cases
+			{
+				cms: []*CheckedMessage{
+					nil,
+					loga.Check(InfoLevel, "apple"),
+				},
+				expectedOK: true,
+				desc:       "nil, A OK",
+				expected: []string{
+					`{"level":"info","msg":"apple","name":"A","i":3}`,
```

Comment:
Now that we're clearing the output buffer for each test, do we still need the loop index in the messages? It'll be easier to add new tests or reorder them later if we omit `i`.